src/vector_query_ops.cpp

```diff
@@ -11,15 +11,14 @@ Option<bool> VectorQueryOps::parse_vector_query_str(const std::string& vector_qu
     // field_name([0.34, 0.66, 0.12, 0.68], exact: false, k: 10)
     size_t i = 0;
     while(i < vector_query_str.size()) {
+        if(vector_query_str[i] == '(' || vector_query_str[i] == '[') {
+            // If we hit a bracket before a colon, it's a missing colon error
+            return Option<bool>(400, "Malformed vector query string: `:` is missing.");
```

Comment:
@tharropoulos Let's make the error string more descriptive: 

```
Malformed vector query string: `:` is missing after the vector field name. 
```

Reply:
Thought about that, thanks. Will update soon

Reply:
83b1b58 addresses it